addOns/domxss/src/main/java/org/zaproxy/zap/extension/domxss/DomXssScanRule.java

```diff
@@ -430,30 +455,54 @@ private void getHelper(WebDriverWrapper wrapper, String url, int retry) {
         return new ArrayList<>();
     }
 
+    private String getAlertDialogText() {
+        try {
+            org.openqa.selenium.Alert alertDialog = driver.getDriver().switchTo().alert();
+            if (alertDialog != null) {
+                return alertDialog.getText();
+            } else {
+                throw new Throwable();
+            }
+        } catch (Throwable e) {
+            // No alert
+            return "";
+        }
+    }
+
     private void scanHelper(String attackVector, String url) throws DomAlertException {
         if (this.isStop()) {
             return;
         }
 
+        String dialogText = "";
         try {
             getHelper(driver, url);
+            dialogText = getAlertDialogText();
```

Comment:
This call does not seem to have effect.

Reply:
Is the whole original block wrong then? If the exception can be thrown and stats counted then there should be circumstances in which we'd need to extract the text.

Reply:
Not sure about the original block, but now it needs changes.

This statement is executed when the dialogue is shown, the text is correctly extracted but then it does no check and tries the following attack (which triggers the exception as the dialogue was not yet closed just the text extracted).

Reply:
This statement is still needed, or use a finally block to the the check. I'd also suggest accept the dialogue after getting the text (to "clean up").